transfer-eth-erc20/to-near.js

```diff
@@ -82,85 +79,43 @@ class TransferETHERC20ToNear {
+    const receipt = await await robustWeb3.getTransactionReceipt(lockedEvent.transactionHash)
+    const block = await robustWeb3.getBlock(lockedEvent.blockNumber)
     const tree = await extractor.buildTrie(block)
     const proof = await extractor.extractProof(
-      web3,
-      block,
+      lockedEvent.blockNumber,
       tree,
       receipt.transactionIndex
     )
-    let txLogIndex = -1
-
-    let logFound = false
-    let log
-    for (let receiptLog of receipt.logs) {
-      txLogIndex++
-      const blockLogIndex = receiptLog.logIndex
-      if (blockLogIndex === lockedEvent.logIndex) {
-        logFound = true
-        log = receiptLog
-        break
-      }
-    }
-    if (logFound) {
-      TransferETHERC20ToNear.recordTransferLog({
-        finished: 'find-proof',
-        proof,
-        log,
-        txLogIndex,
-        receipt,
-        lockedEvent,
-        block,
-      })
-    } else {
+    const log = receipt.logs.find(l => l.logIndex === lockedEvent.logIndex)
+    if (!log) {
       console.log(`Failed to find log for event ${lockedEvent}`)
       TransferETHERC20ToNear.showRetryAndExit()
     }
-  }
-
-  static async waitBlockSafe({
-    log,
-    proof,
-    receipt,
-    txLogIndex,
-    lockedEvent,
-    block,
-    ethOnNearClientContract,
-  }) {
-    const log_entry_data = logFromWeb3(log).serialize()
-    const receipt_index = proof.txIndex
-    const receipt_data = receiptFromWeb3(receipt).serialize()
-    const header_data = proof.header_rlp
-    const _proof = []
-    for (const node of proof.receiptProof) {
-      _proof.push(utils.rlp.encode(node))
-    }
-
-    const proof_locker = {
-      log_index: txLogIndex,
-      log_entry_data: log_entry_data,
-      receipt_index: receipt_index,
-      receipt_data: receipt_data,
-      header_data: header_data,
-      proof: _proof,
+    return {
+      log_index: lockedEvent.logIndex,
+      log_entry_data: Log.fromWeb3(log).serialize(),
+      receipt_index: proof.txIndex,
+      receipt_data: Receipt.fromWeb3(receipt).serialize(),
+      header_data: proof.header_rlp,
+      proof: proof.receiptProof.map(utils.rlp.encode),
     }
+  }
 
-    const new_owner_id = lockedEvent.returnValues.accountId
-    const amount = lockedEvent.returnValues.amount
-    console.log(
-      `Transferring ${amount} tokens from ${lockedEvent.returnValues.token} ERC20. From ${lockedEvent.returnValues.sender} sender to ${new_owner_id} recipient`
-    )
-
-    const blockNumber = block.number
+  static async waitBlockSafe({ lockedEvent, ethOnNearClientContract }) {
     // Wait until client accepts this block number.
     while (true) {
       // @ts-ignore
       const last_block_number = (
         await ethOnNearClientContract.last_block_number()
       ).toNumber()
+      // Why should the client decide what's safe?
+      // Shouldn't MintableFungibleToken enforce this?
+      // And a frontend set expectations accordingly?
```

Comment:
@ailisp and these questions? To me it seems like we should:

* set `25` in MintableFungibleToken
* adjust expectations accordingly here
* never call `ethOnNearClientContract.block_hash_safe`

Reply:
MintableFungibleToken does enforce this, so if frontend doesn't wait block_hash_safe and continue to next step(mint), it will be rejected by MintableFungibleToken. So we also wait in frontend

Reply:
where is `25` comes from?

Reply:
`25` is mentioned in the blog post introducing Bridge (https://near.org/blog/eth-near-rainbow-bridge/). I would expect it's part of `MintableFungibleToken`. I did _not_ expect it to be part of EthOnNearClient, but the existence of `ethOnNearClientContract.block_hash_safe` indicates otherwise. 

Sounds like I could get away with never calling `block_hash_safe` on the frontend, if the frontend makes the same assumptions.

Reply:
However, here in `rainbow-bridge-lib`, I think it's worth checking both

Reply:
> Sounds like I could get away with never calling block_hash_safe on the frontend, if the frontend makes the correct assumptions.

Yes if contract is configured so. Need to check with @nearmax if it's really 25 in our deployment, it's probably a bigger number.